p2p/host/peerstore/pstoremem/addr_book.go

```diff
@@ -114,11 +114,12 @@ func (pa *peerAddrs) NextExpiry() time.Time {
 	if len(pa.expiringHeap) == 0 {
 		return time.Time{}
 	}
-	return pa.expiringHeap[len(pa.expiringHeap)-1].Expires
+	return pa.expiringHeap[0].Expires
```

Comment:
Are you sure this is right? The min value should be at the end, no?

Reply:
Yes, I added a couple of tests. See: `TestPeerAddrsNextExpiry`

If you're confused, as I was, by the fact that `heap.Interface.Pop` requires removal from the end, that's because `heap.Pop` swaps the first and last element before calling `heap.Interface.Pop`. 

from container/heap/heap.go
```
// Pop removes and returns the minimum element (according to Less) from the heap.
// The complexity is O(log n) where n = h.Len().
// Pop is equivalent to [Remove](h, 0).
func Pop(h Interface) any {
	n := h.Len() - 1
	h.Swap(0, n)
	down(h, 0, n)
	return h.Pop()
}
```

Reply:
ah, thanks. That makes sense.